gauntlet/packages/gauntlet-serum-multisig/src/commands/multisig.ts

```diff
@@ -0,0 +1,140 @@
+import { Result, WriteResponse } from '@chainlink/gauntlet-core'
+import { TransactionResponse, SolanaCommand, RawTransaction } from '@chainlink/gauntlet-solana'
+
+import { logger } from '@chainlink/gauntlet-core/dist/utils'
+import { PublicKey, SYSVAR_RENT_PUBKEY, Keypair } from '@solana/web3.js'
+import { CONTRACT_LIST, getContract } from '@chainlink/gauntlet-solana-contracts'
+
+// enum ACTIONS {
+//   create = 'create',
+//   approve = 'approve',
+//   execute = 'execute',
+// }
+
+export const wrapCommand = (command) => {
+  return class Multisig extends SolanaCommand {
+    command: SolanaCommand
+    multisigAddress
+    multisig
+    address
+    program
+    rawTx
+
+    static id = `${command.id}`
+
+    constructor(flags, args) {
+      super(flags, args)
+      logger.info(`Serum multisig wrapping of ${command.id}`)
+
+      this.command = new command(flags, args)
+      this.command.invokeMiddlewares(this.command, this.command.middlewares)
+    }
+
+    createProposal = async (rawTx: RawTransaction) => {
+      logger.loading(`Creating proposal`)
+      const txSize = 1000
+      const transaction = Keypair.generate()
+      const txPublicKey = transaction.publicKey
+      const tx = await this.program.rpc.createTransaction(rawTx.programId, rawTx.accounts, rawTx.data, {
+        accounts: {
+          multisig: this.multisigAddress,
+          transaction: txPublicKey,
+          proposer: this.wallet.payer.publicKey,
+          rent: SYSVAR_RENT_PUBKEY,
+        },
+        instructions: [await this.program.account.transaction.createInstruction(transaction, txSize)],
+        signers: [transaction, this.wallet.payer],
+      })
+      return { tx, txPublicKey }
+    }
+
+    approveProposal = async (txAccount) => {
+      logger.loading(`Approving proposal`)
+      const tx = await this.program.rpc.approve({
+        accounts: {
+          multisig: this.multisigAddress,
+          transaction: txAccount,
+          owner: this.wallet.publicKey,
+        },
+      })
+      return tx
+    }
+
+    executeProposal = async (txAccount) => {
```

Comment:
let's rename this to either `proposal` or `tx`, and it should be a `PublicKey` type